backend/routers/spiritual.py

```
@@ -552,11 +552,16 @@ async def get_spiritual_progress(user_id: str, request: Request):
         raise HTTPException(status_code=401, detail="Authentication required")
     
     try:
+        # Convert user_id to integer if needed for database queries
+        try:
+            user_id_int = int(user_id)
+        except (ValueError, TypeError):
+            user_id_int = None
```

Ruff (0.11.9)

Local variable `user_id_int` is assigned to but never used

Remove assignment to unused variable `user_id_int`

(F841)